src/pgexporter_ext/lib.c

```diff
@@ -485,17 +485,20 @@ os_info(Tuplestorestate* tupstore, TupleDesc tupdesc)
       nulls[OS_INFO_DOMAIN_NAME] = true;
    }
 
-   os_info_file = fopen("/etc/system-release", "r");
+   os_info_file = fopen("/etc/os-release", "r");
 
    if (!os_info_file)
    {
       nulls[OS_INFO_NAME] = true;
    }
    else
    {
-      if (fgets(&buffer[0], max_length, os_info_file) != NULL)
+      while (fgets(&buffer[0], max_length, os_info_file) != NULL)
       {
-         memcpy(os_name, buffer, strlen(buffer) - 1);
+         if(strstr(buffer, "PRETTY_NAME") != NULL){
```

Comment:
Use `uncrustify` for code formatting. Instructions are [here](https://github.com/pgexporter/pgexporter/blob/main/CONTRIBUTING.md#code-style). Also, please look at the other instructions as well, including the one about squashing.

Reply:
Two things:
- The config file contains some deprecated options which uncrustify complains about
```
❯ ./uncrustify.sh
uncrustify.cfg:472: option 'sp_type_question' is deprecated; did you want to use 'sp_before_ptr_star' instead?
Option<NUM>: at uncrustify.cfg:1182: Expected number , for 'indent_shift'; got 'false'
Option<NUM>: at uncrustify.cfg:1356: Expected number , for 'indent_comma_brace'; got 'false'
Option<NUM>: at uncrustify.cfg:1360: Expected number , for 'indent_comma_paren'; got 'false'
Option<NUM>: at uncrustify.cfg:1364: Expected number , for 'indent_bool_paren'; got 'false'
uncrustify.cfg:2225: option 'nl_func_var_def_blk' is deprecated; it has been replaced by 'nl_var_def_blk_end_func_top'.
You can also use 'nl_var_def_blk_end' for additional functionality
Option<UNUM>: at uncrustify.cfg:2741: Expected unsigned number , for 'align_nl_cont'; got 'false'
Option<UNUM>: at uncrustify.cfg:2988: Expected unsigned number , for 'mod_full_brace_if_chain'; got 'false'
uncrustify.cfg:3156: option 'pp_space' is deprecated; it has been replaced by 'pp_space_after'.
Option<NUM>: at uncrustify.cfg:3217: Expected number , for 'pp_indent_brace'; got 'true'
uncrustify.cfg:472: option 'sp_type_question' is deprecated; did you want to use 'sp_before_ptr_star' instead?
Option<NUM>: at uncrustify.cfg:1182: Expected number , for 'indent_shift'; got 'false'
Option<NUM>: at uncrustify.cfg:1356: Expected number , for 'indent_comma_brace'; got 'false'
Option<NUM>: at uncrustify.cfg:1360: Expected number , for 'indent_comma_paren'; got 'false'
Option<NUM>: at uncrustify.cfg:1364: Expected number , for 'indent_bool_paren'; got 'false'
uncrustify.cfg:2225: option 'nl_func_var_def_blk' is deprecated; it has been replaced by 'nl_var_def_blk_end_func_top'.
You can also use 'nl_var_def_blk_end' for additional functionality
Option<UNUM>: at uncrustify.cfg:2741: Expected unsigned number , for 'align_nl_cont'; got 'false'
Option<UNUM>: at uncrustify.cfg:2988: Expected unsigned number , for 'mod_full_brace_if_chain'; got 'false'
uncrustify.cfg:3156: option 'pp_space' is deprecated; it has been replaced by 'pp_space_after'.
Option<NUM>: at uncrustify.cfg:3217: Expected number , for 'pp_indent_brace'; got 'true'
uncrustify.cfg:472: option 'sp_type_question' is deprecated; did you want to use 'sp_before_ptr_star' instead?
Option<NUM>: at uncrustify.cfg:1182: Expected number , for 'indent_shift'; got 'false'
Option<NUM>: at uncrustify.cfg:1356: Expected number , for 'indent_comma_brace'; got 'false'
Option<NUM>: at uncrustify.cfg:1360: Expected number , for 'indent_comma_paren'; got 'false'
Option<NUM>: at uncrustify.cfg:1364: Expected number , for 'indent_bool_paren'; got 'false'
uncrustify.cfg:2225: option 'nl_func_var_def_blk' is deprecated; it has been replaced by 'nl_var_def_blk_end_func_top'.
You can also use 'nl_var_def_blk_end' for additional functionality
Option<UNUM>: at uncrustify.cfg:2741: Expected unsigned number , for 'align_nl_cont'; got 'false'
Option<UNUM>: at uncrustify.cfg:2988: Expected unsigned number , for 'mod_full_brace_if_chain'; got 'false'
uncrustify.cfg:3156: option 'pp_space' is deprecated; it has been replaced by 'pp_space_after'.
Option<NUM>: at uncrustify.cfg:3217: Expected number , for 'pp_indent_brace'; got 'true'
uncrustify.cfg:472: option 'sp_type_question' is deprecated; did you want to use 'sp_before_ptr_star' instead?
Option<NUM>: at uncrustify.cfg:1182: Expected number , for 'indent_shift'; got 'false'
Option<NUM>: at uncrustify.cfg:1356: Expected number , for 'indent_comma_brace'; got 'false'
Option<NUM>: at uncrustify.cfg:1360: Expected number , for 'indent_comma_paren'; got 'false'
Option<NUM>: at uncrustify.cfg:1364: Expected number , for 'indent_bool_paren'; got 'false'
uncrustify.cfg:2225: option 'nl_func_var_def_blk' is deprecated; it has been replaced by 'nl_var_def_blk_end_func_top'.
You can also use 'nl_var_def_blk_end' for additional functionality
Option<UNUM>: at uncrustify.cfg:2741: Expected unsigned number , for 'align_nl_cont'; got 'false'
Option<UNUM>: at uncrustify.cfg:2988: Expected unsigned number , for 'mod_full_brace_if_chain'; got 'false'
uncrustify.cfg:3156: option 'pp_space' is deprecated; it has been replaced by 'pp_space_after'.
Option<NUM>: at uncrustify.cfg:3217: Expected number , for 'pp_indent_brace'; got 'true'
```
I used the config from pgmoneta to fix this.

- Even after fixing it, it does not change anything in the file. Is the formatting correct or uncrustify not working correctly?

Reply:
I have updated uncrustify

Reply:
> I have updated uncrustify

So there should not be any error?

Reply:
@jesperpedersen I think you updated it in pgexporter main repo

Reply:
Done

Reply:
The main issue is wrong directory name in uncrustify.sh: https://github.com/pgexporter/pgexporter_ext/blob/a6a37d6e21b312ffa3ae7c7c57dbd9c469723afb/uncrustify.sh#L13

It should be `src/pgexporter_ext/*.c`

Should I fix this in this PR?

Reply:
Good catch - done

Reply:
Rebased.